sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/owaspzapwrapper/cli/OwaspZapScanExecutor.java

```diff
@@ -39,15 +38,29 @@ public void execute(OwaspZapScanContext scanContext) throws ZapWrapperRuntimeExc
 
     }
 
+    private void assertTargetReachable(OwaspZapScanContext scanContext) {
+        boolean isReachable = false;
+        for (String url : scanContext.getOwaspZapURLsIncludeList()) {
+            URI uriToCheck = URI.create(url);
+            if (connectionChecker.isTargetReachable(uriToCheck, scanContext.getProxyInformation())) {
+                isReachable = true;
+                break;
```

Comment:
I would use a while loop instead of a for loop and put the break condition in the loop statement:

~~~
while( urls.hasNext() && isReachable != true) {
…
}
~~~